tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/DetectSigningIdentityTaskTests.cs

```diff
@@ -0,0 +1,137 @@
+<dict>
+</dict>
+</plist>";
+
+		const string EmptyEntitlements4 = @"<?xml version=""1.0"" encoding=""UTF-8""?>
+<!DOCTYPE plist PUBLIC ""-//Apple//DTD PLIST 1.0//EN"" ""http://www.apple.com/DTDs/PropertyList-1.0.dtd"">
+<plist version=""1.0"">
+<dict>
+<!-- comment here! -->
+</dict>
+</plist>";
+
+		const string NonEmptyEntitlements1 = @"<?xml version=""1.0"" encoding=""UTF-8""?>
+<!DOCTYPE plist PUBLIC ""-//Apple//DTD PLIST 1.0//EN"" ""http://www.apple.com/DTDs/PropertyList-1.0.dtd"">
+<plist version=""1.0"">
+<dict>
+    <key>com.apple.security.app-sandbox</key>
+    <true/>
+</dict>
+</plist>";
+
+		public class EntitlementTestCase
+		{
+			public string Name = string.Empty;
+			public string Entitlements = string.Empty;
+			public bool Required;
+			public bool IsSimulator;
+			public bool? CodesignRequireProvisioningProfile;
+
+			public override string ToString ()
+			{
+				return Name;
+			}
+		}
+
+		static EntitlementTestCase[] GetEntitlementsTestCases ()
+		{
+			return new EntitlementTestCase[]
+			{
+				new EntitlementTestCase { Name = nameof (EmptyEntitlements1), Entitlements = EmptyEntitlements1, Required = false, IsSimulator = true },
+				new EntitlementTestCase { Name = nameof (EmptyEntitlements2), Entitlements = EmptyEntitlements2, Required = false, IsSimulator = true },
+				new EntitlementTestCase { Name = nameof (EmptyEntitlements3), Entitlements = EmptyEntitlements3, Required = false, IsSimulator = true },
+				new EntitlementTestCase { Name = nameof (EmptyEntitlements4), Entitlements = EmptyEntitlements4, Required = false, IsSimulator = true },
+				new EntitlementTestCase { Name = nameof (NonEmptyEntitlements1), Entitlements = NonEmptyEntitlements1, Required = true, IsSimulator = true },
+				new EntitlementTestCase { Name = nameof (EmptyEntitlements1) + "_Required", Entitlements = EmptyEntitlements1, Required = true, IsSimulator = true, CodesignRequireProvisioningProfile = true },
+				new EntitlementTestCase { Name = nameof (NonEmptyEntitlements1) + "_NotRequired", Entitlements = NonEmptyEntitlements1, Required = false, IsSimulator = true, CodesignRequireProvisioningProfile = false },
+			};
+		}
+
+		[Test]
+		[TestCaseSource (nameof (GetEntitlementsTestCases))]
+		public void EmptyEntitlements (EntitlementTestCase testCase)
+		{
+			var dir = Cache.CreateTemporaryDirectory ();
+			var entitlementsPath = Path.Combine (dir, "Entitlements.plist");
+			File.WriteAllText (entitlementsPath, testCase.Entitlements);
+
+			var task = CreateTask (dir, simulator: testCase.IsSimulator);
+			task.CodesignEntitlements = entitlementsPath;
+			task.SdkIsSimulator = testCase.IsSimulator;
+			if (testCase.CodesignRequireProvisioningProfile.HasValue)
+				task.CodesignRequireProvisioningProfile = testCase.CodesignRequireProvisioningProfile.Value.ToString ();
+
+			ExecuteTask (task);
+
+			if (testCase.Required) {
+				Assert.That (task.DetectedAppId, Is.Not.Null.And.No.Empty, "DetectedAppId");
```

Comment:
```suggestion
				Assert.That (task.DetectedAppId, Is.Not.Null.And.Not.Empty, "DetectedAppId");
```
I believe this is a typo